plugins/colorhug/fu-colorhug-device.c

```diff
@@ -423,9 +423,13 @@ fu_colorhug_device_write_firmware (FuDevice *device,
 				   GError **error)
 {
 	FuColorhugDevice *self = FU_COLORHUG_DEVICE (device);
+	FuProgress *progress = fu_device_get_progress_helper(device);
 	g_autoptr(GBytes) fw = NULL;
 	g_autoptr(GPtrArray) chunks = NULL;
 
+	/* progress */
+	fu_progress_set_custom_steps(progress, 19 /* erase */, 45 /* write */, 36 /* read */, -1);
```

Comment:
so in my mind I was thinking that every time the device entered into a certain state (say erase or write) you were going to be able to map out a new step.  Is that how you're thinking too?  And things with multiple partitions or multiple erase steps as long as there is a state change it keeps working?

IE set status to erase, write, erase, write, read would be 5 steps

Reply:
Yes, that's exactly what I think this does; i.e. for 2 partitions each with E,W,R you'd have one parent progress with two steps, and then two children each with three steps, so you've got 6 non-similar "splits" in your 0..100% scale.